### PR TITLE
AK-14181 Add AWS-VPC routing options support to resource_alkira_conne…

### PR DESCRIPTION
**Pull request by spikewang**
…ctor_aws_vpc

Adding routing options support to resource_alkira_connector_aws_vpc. With this
commit, either vpc_cidr or vpc_subnets is needed for creating AWS-VPC connector.

+ add new var `vpc_cidr` to specify AWS-VPC CIDR.
+ Add new var `vpc_subnets` to specifcy AWS-VPC subnets.
+ Add new set `vpc_routetable` to specify AWS-VPC route tables and prefixs.

### CODE REVIEW

**Comment by spikewang** [comment]
Now with AWS-VPC connector, we will need either `vpc_cidr` or `vpc_subnets` as part of the input. If you have older configuration, error message will be displayed during `terraform plan` or `terraform apply` to remind you this.

To use the new route table section:

```
  vpc_route_table {
    id              = "rtb-XXXX"
    prefix_list_ids = [data.alkira_policy_prefix_list.test1.id]
    options         = "ADVERTISE_CUSTOM_PREFIX"
  }
```

File: alkira/resource_alkira_connector_aws_vpc.go [hunk 1]
```diff
@@ -11,77 +11,134 @@ import (
...
+							Optional:    true,
+						},
+						"prefix_list_ids": {
+							Description: "Prefix List Ids",
+							Type:        schema.TypeList,
+							Optional:    true,
+							Elem:        &schema.Schema{Type: schema.TypeInt},
+						},
+
+						"options": {
```
Comment (rvichery) [inline alkira/resource_alkira_connector_aws_vpc.go:103]: We should probably add:
```
ValidateFunc: validation.StringInSlice([]string{"ADVERTISE_DEFAULT_ROUTE", OVERRIDE_DEFAULT_ROUTE", "ADVERTISE_CUSTOM_PREFIX"}, false),
```

**Review by rvichery — CHANGES_REQUESTED** [review]
Couple of small things to fix for better validation.
Also I am wondering if we should use TypeList for `billing_tags`, `vpc_subnets` and `prefix_list_ids` as they are not ordered collection of items AFAIK.

File: alkira/resource_alkira_connector_aws_vpc.go (hunk 1 above)
Comment (spikewang) [reply alkira/resource_alkira_connector_aws_vpc.go:103]: Yep, about to do that.

File: alkira/helper.go [hunk 2]
```diff
@@ -185,35 +186,62 @@ func expandPolicyRuleListRules(in *schema.Set) []alkira.PolicyRuleListRule {
...
+
+// generateUserInputPrefixes generate UserInputPrefixes used in AWS-VPC connector
+func generateUserInputPrefixes(cidr string, subnets []interface{}) ([]alkira.InputPrefixes, error) {
+
+	if cidr == "" && subnets == nil {
+		return nil, fmt.Errorf("ERROR: either vpc_subnets and vpc_cidr should be specified.")
+	}
+
+	if cidr != "" && len(subnets) > 0 {
+		return nil, fmt.Errorf("ERROR: vpc_subnets and vpc_cidr can't be specified at the same time.")
```
[the remark this answers is not in the document]
Comment (spikewang) [reply alkira/helper.go:222]: Let me check

**Review by sahilthapar1234 — APPROVED** [review]

**Comment by spikewang** [comment]
Yep, we could, will do those changes in a separate one. Couple more changes are pending..

**Pull request merged**